File: Test-SB/src/main/java/com/test/entities/Produit.java
package com.test.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "produits")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Produit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "product_name", nullable = false)
    private String productName;

    @Column(name = "description")
    private String description;

    @Column(name = "prix", nullable = false)
    private int prix;

    @Column(name = "quantity")
    private int quantity;

    @ManyToOne
    @JoinColumn(name = "categories_id", nullable = false)
    private Categorie categorie;

}

File: src/app/models/produit.ts
import { Categorie } from './categorie';

/*
export interface Produit {
  id: number;
  productName: string;
  description: string;
  prix: number;
  quantity: number;
  categories_id: number;
  categoryName?: string; // Pour afficher le nom de la catégorie dans la liste des produits
}

 */

export class Produit {
  id!: number;
  productName!: string;
  description!: string;
  prix!: number;
  quantity!: number;
  categorie!: Categorie;
}


